homeassistant/components/livisi/binary_sensor.py

```diff
@@ -0,0 +1,155 @@
+"""Code to handle a Livisi Binary Sensor."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    DOMAIN,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+    PSS_DEVICE_TYPE,
+    WDS_DEVICE_TYPE,
+)
+from .coordinator import LivisiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up binary_sensor device."""
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add binary_sensor."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        entities: list[BinarySensorEntity] = []
+
+        """ Added Window Sensor """
+        for device in shc_devices:
+            if device["type"] == "WDS" and device["id"] not in coordinator.devices:
+                livisi_binary: BinarySensorEntity = create_entity(
+                    config_entry, device, coordinator
+                )
+                LOGGER.debug("Include device type: %s", device["type"])
+                coordinator.devices.add(device["id"])
+                entities.append(livisi_binary)
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        coordinator.async_add_listener(handle_coordinator_update)
+    )
+
+
+def create_entity(
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> BinarySensorEntity:
+    """Create Binary Sensor Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    room_id: str = device["location"]
+    room_name: str = coordinator.rooms[room_id]
+    livisi_binary = LivisiWindowSensor(
+        config_entry,
+        coordinator,
+        unique_id=device["id"],
```

Comment:
Is the ID unique to the window sensor?

Reply:
Yes the id is unique

![2022-11-30 12_34_56](https://user-images.githubusercontent.com/78791840/204788131-34170826-4d51-4922-8834-17a7078f752b.png)

Reply:
Right, but there is a chance there will be a need to create multiple binary sensors in home assistant for a livisi window sensor in the future. For example, maybe we want to add a "low battery" binary sensor?

If that's the case, we should consider formatting the unique_id to avoid future collisions, for example:
`unique_id=f"{device['id']_window_sensor"`, or whatever is appropriate.

We could also leave it as is, and just use the ID as is for the "primary" function of the device.

Reply:
You are right. I have not yet thought of this possibility. But with the window sensor of livisi/rwe is not known to me that there is something like battery status or brightness. But I have added a postfix the id.